Added failOnStdError input (#61)
Azure/arm-deploy: README.md

* `parameters`: Specify the path or URL to the Azure Resource Manager deployment parameter values. Or local / remote value file.  
* `deploymentMode`: `Incremental`(default) (only add resources to resource group) or `Complete` (remove extra resources from resource group) or `Validate` (only validates the template). 
* `deploymentName`: Specifies the name of the resource group deployment to create.

> 

Reviewer: Add an example for stderr and explain what it means via the example

> * `failOnStdError`: Specify whether to fail the action when std error is not null.

Reviewer: Rename to failOnStdErr as stderr is more easily recognised term

Reviewer: Use that in description as well

> * `failOnStdError`: Specify whether to fail the action when std error is not null.

Reviewer: Specify the default value

> * `failOnStdError`: Specify whether to fail the action when std error is not null.

Reviewer: Rather than saying not null, say if the some data is written to stderr stream of az cli

ARM Deploy Actions is supported for the Azure public cloud as well as Azure government clouds ('AzureUSGovernment' or 'AzureChinaCloud') and Azure Stack ('AzureStack') Hub. Before running this action, login to the respective Azure Cloud  using [Azure Login](https://github.com/Azure/login) by setting appropriate value for the `environment` parameter.

## Example on how to use failOnStdErr
In this example, we are setting `failOnStdErr` to false. 

```yaml
- uses: azure/arm-deploy@v1
  id: deploy
  with:
    resourceGroupName: azurearmaction
    template: examples/template/template.json
    parameters: examples/template/parameters.json
    deploymentName: github-advanced-test
    failOnStdErr: false
```

> `failOnStdErr` equals true means that if there is some data written to stdErr and return code from az-cli is 0 then we will fail the action. 

Reviewer: ```suggestion
`failOnStdErr` equals true implied that if some data is written to stdErr and exit code from az-cli is 0, then action execution will fail. 
```

> The default value of `failOnStdErr` is true to support back compatibility. `failOnStdErr` equals true means that if there is some data written to stdErr and return code from az-cli is 0 then we will fail the action. 

Reviewer: Remove this `The default value of `failOnStdErr` is true to support back compatibility.`

> `failOnStdErr` equals false means that if there is some data written to stdErr and return code from az-cli is 0 then action will be return a success. This input is added to support cases where stdErr is being used to stream warnings. 

Reviewer: ```suggestion
`failOnStdErr` equals false implies that if some data is written to stdErr and return code from az-cli is 0, then action will continue execution. This input is added to support cases where stdErr is being used to stream warning or progress info. 
```

> Return code 1 will always lead to failure of action irrespective the value of `failOnStdErr`.

Reviewer: ```suggestion
Non zero Exit code will always lead to failure of action irrespective the value of `failOnStdErr`.
```
